src/tests/suite/daos_test_common.c

```diff
@@ -1026,3 +1026,186 @@ get_daos_prop_with_user_acl_perms(uint64_t perms)
 	D_FREE(user);
 	return prop;
 }
+
+int
+get_pid_of_process(char *host, char *dpid, char *proc)
+{
+	char    command[256];
+	size_t  len = 0;
+	size_t  read;
+	char    *line = NULL;
+
+	snprintf(command, sizeof(command),
+		 "ssh %s pgrep %s", host, proc);
+	FILE *fp1 = popen(command, "r");
+
+	print_message("Command= %s\n", command);
+	if (fp1 == NULL)
+		return -DER_INVAL;
+
+	while ((read = getline(&line, &len, fp1)) != -1) {
+		print_message("%s pid = %s", proc, line);
+		strcat(dpid, line);
+	}
+
+	pclose(fp1);
+	return 0;
+}
+
+int
+get_server_config(char *host, char *server_config_file)
+{
+	char	command[256];
+	size_t	len = 0;
+	size_t	read;
+	char	*line = NULL;
+	char	*pch;
+	char    *dpid;
+	int	rc;
+	char    daos_proc[16] = "daos_server";
+
+	D_ALLOC(dpid, 16);
+	rc = get_pid_of_process(host, dpid, daos_proc);
+	assert_int_equal(rc, 0);
+
+	snprintf(command, sizeof(command),
+		 "ssh %s ps ux -A | grep %s", host, dpid);
+	FILE *fp = popen(command, "r");
+
+	print_message("Command %s", command);
+	if (fp == NULL)
+		return -DER_INVAL;
```

Comment:
It would be good to print the stderror(errno) here or something so we have a clue why it failed.

Reply:
Popen internally performed by shell so any failure error will be printed on console.
It will return NULL in case of fork(), pipe() or memory allocation failed which is just to check it's not NULL in any case before reading it through in case of command success.